File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;

    int prev_num = arr[0];
    answer.push_back(prev_num);
    for (int i = 1; i < arr.size(); i++) {
        if (prev_num != arr[i]) {
            answer.push_back(arr[i]);
            prev_num = arr[i];
        }
    }

    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        ArrayList<Integer> temp= new ArrayList<>();
        Stack<Integer> stack = new Stack<>();
        stack.push(Integer.valueOf(arr[0]));
        temp.add(Integer.valueOf(arr[0]));
        
        for (int i = 1; i < arr.length; i++) {
            if (!stack.empty()) {
                if (stack.peek() != arr[i]) {
                    stack.push(Integer.valueOf(arr[i]));
                    temp.add(Integer.valueOf(arr[i]));
                }
            } 
        }
        
        int[] answer = new int[temp.size()];
        
        for (int i = 0; i < temp.size(); i++) {
            answer[i] = temp.get(i);
        }
        return answer;
    }
}